v2/spanner-common/src/main/java/com/google/cloud/teleport/v2/spanner/migrations/metadata/CassandraSourceMetadata.java

```diff
@@ -0,0 +1,186 @@
+   * @param resultSet The ResultSet containing schema information.
+   */
+  public static void generateSourceSchema(Schema schema, ResultSet resultSet) {
+    SourceSchema sourceSchema = generateSourceSchema(resultSet);
+    Map<String, com.google.cloud.teleport.v2.spanner.migrations.schema.SourceTable> sourceTableMap =
+        convertSourceSchemaToMap(sourceSchema);
+    schema.setSrcSchema(sourceTableMap);
+    schema.setToSource(convertSourceToNameAndColsTable(sourceSchema.tables().values()));
+  }
+
+  /**
+   * Converts a SourceSchema to a map of Spanner table names to SourceTable objects.
+   *
+   * @param sourceSchema The SourceSchema to convert.
+   * @return A map where the key is the table name and the value is the corresponding SourceTable.
+   */
+  private static Map<String, com.google.cloud.teleport.v2.spanner.migrations.schema.SourceTable>
+      convertSourceSchemaToMap(SourceSchema sourceSchema) {
+    return sourceSchema.tables().entrySet().stream()
+        .collect(
+            Collectors.toMap(
+                Map.Entry::getKey, entry -> convertSourceTableToSourceTable(entry.getValue())));
+  }
+
+  /**
+   * Converts a SourceTable to a SourceTable for Spanner.
+   *
+   * @param sourceTable The SourceTable to convert.
+   * @return A converted SourceTable object.
+   */
+  private static com.google.cloud.teleport.v2.spanner.migrations.schema.SourceTable
+      convertSourceTableToSourceTable(SourceTable sourceTable) {
+    List<SourceColumn> columns = sourceTable.columns();
+
+    String[] colIds = columns.stream().map(SourceColumn::name).toArray(String[]::new);
+
+    Map<String, SourceColumnDefinition> colDefs =
+        columns.stream()
+            .collect(
+                Collectors.toMap(
+                    SourceColumn::name,
+                    col ->
+                        new SourceColumnDefinition(
+                            col.name(),
+                            new SourceColumnType(col.sourceType(), new Long[0], new Long[0]))));
+
+    ColumnPK[] primaryKeys =
+        columns.stream()
+            .filter(SourceColumn::isPrimaryKey)
+            .map(col -> new ColumnPK(col.name(), getPrimaryKeyOrder(col)))
+            .toArray(ColumnPK[]::new);
+
+    return new com.google.cloud.teleport.v2.spanner.migrations.schema.SourceTable(
+        sourceTable.name(),
+        SCHEMA_NAME, // Use constant schema name
+        colIds,
+        colDefs,
+        primaryKeys);
+  }
+
+  /**
+   * Determines if a column is a primary key.
+   *
+   * @param kind The column kind.
+   * @return true if the column is a primary key, false otherwise.
+   */
+  private static boolean isPrimaryKey(String kind) {
+    return "partition_key".equals(kind) || "clustering".equals(kind);
+  }
+
+  /**
+   * Gets the order of the primary key column.
+   *
+   * @param col The SourceColumn.
+   * @return The order of the primary key.
+   */
+  private static int getPrimaryKeyOrder(SourceColumn col) {
```

Comment:
what is this order used for ? It is not very obvious here

Reply:
Yes , Correct @bharadwaj-aditya but we have added to align with structure , For a Note: we are not using any where the order